File: c++/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> M;
        vector<int> result;
        for(int i=0;i<nums.size();i++){
            if(M[target - nums[i]]){
                result.push_back(M[target - nums[i]] - 1);
                result.push_back(i);
            }
            else
                M[nums[i]] = i + 1;
        }
        return result;
    }
};

File: java/1. Two sum.java
class Solution {
    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numberToPosition = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (numberToPosition.containsKey(target - nums[i])) {
                int position = numberToPosition.get(target - nums[i]);
                return new int[] { position, i };
            }

            numberToPosition.put(nums[i], i);
        }

        return new int[] { -1, -1 };
    }
}